api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TXCurrQueries.java

```diff
@@ -0,0 +1,309 @@
+            + "and e.location_id = :location and e.encounter_datetime <= :onOrBefore "
+            + "group by p.patient_id";
+
+    return String.format(
+        query,
+        adultoSeguimentoEncounterType,
+        masterCardEncounterType,
+        stateOfStayPriorArtPatientConcept,
+        patientHasDiedConcept);
+  }
+
+  public static String getTransferredOutPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate(
+      int adultoSeguimentoEncounterType,
+      int masterCardEncounterType,
+      int stateOfStayPriorArtPatientConcept,
+      int transferredOutConcept) {
+
+    String query =
+        "select  p.patient_id "
+            + "from patient p "
+            + "inner join encounter e on e.patient_id=p.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where e.encounter_type in (%s,%s) and p.voided=0  and e.voided=0 and o.voided=0 "
+            + "and o.concept_id=%s and   o.value_coded=%s "
+            + "and e.location_id = :location and e.encounter_datetime <= :onOrBefore "
+            + "group by p.patient_id ";
+
+    return String.format(
+        query,
+        adultoSeguimentoEncounterType,
+        masterCardEncounterType,
+        stateOfStayPriorArtPatientConcept,
+        transferredOutConcept);
+  }
+
+  public static String getPatientSuspendedInFichaResumeAndClinicaOfMasterCardByReportEndDate(
+      int adultoSeguimentoEncounterType,
+      int masterCardEncounterType,
+      int stateOfStayPriorArtPatientConcept,
+      int suspendedTreatmentConcept) {
+    String query =
+        "select  p.patient_id "
+            + "from patient p "
+            + "inner join encounter e on e.patient_id=p.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where e.encounter_type in (%s,%s) and p.voided=0  and e.voided=0 and o.voided=0 "
+            + "and o.concept_id=%s and   o.value_coded=%s "
+            + "and e.location_id = :location and e.encounter_datetime <= :onOrBefore "
+            + "group by p.patient_id ";
+
+    return String.format(
+        query,
+        adultoSeguimentoEncounterType,
+        masterCardEncounterType,
+        stateOfStayPriorArtPatientConcept,
+        suspendedTreatmentConcept);
+  }
+
+  public static String getPatientWhoAfterMostRecentDateHaveDrusPickupOrConsultation(
+      int adultoSeguimentoEncounterType,
+      int aRVPediatriaSeguimentoEncounterType,
+      int aRVPharmaciaEncounterType,
+      int masterCardDrugPickupEncounterType,
+      int artDatePickup) {
+    String query =
+        " select p.patient_id "
+            + " from patient p "
+            + " inner join encounter e on  e.patient_id = p.patient_id "
+            + " where  p.voided=0  and e.voided=0 "
+            + " and e.encounter_type in (%s,%s,%s)  "
+            + " and e.encounter_datetime = (select  max(encounter_datetime) from encounter where  patient_id= p.patient_id) "
```

Comment:
So the specs say: `encounter_datetime> the most recent date` 

this query will pick the most recent encounter, but not necessarily the most recent encounter for those encounter types. Have you confirmed with Pinki if we need to be considering any/all encounters or just those linked to those 3 encounter types?

Reply:
@cfaife I see the max encounter date query is still not filtering on encounter type - have you confirmed with Pinki if this is ok?

Reply:
done